protostar/cli/signer_factory.py

```diff
@@ -0,0 +1,97 @@
+import importlib
+import os
+from pathlib import Path
+from typing import Optional
+
+from starknet_py.net.models import StarknetChainId
+from starknet_py.net.signer import BaseSigner
+
+from protostar.cli import Command
+from protostar.protostar_exception import ProtostarException
+from protostar.starknet_gateway import AccountConfig, create_stark_curve_signer
+
+PRIVATE_KEY_ENV_VAR_NAME = "PROTOSTAR_ACCOUNT_PRIVATE_KEY"
+
+
+SIGNABLE_ARGUMENTS = [
+    Command.Argument(
+        name="account-address",
+        description="Account address",
+        type="str",
+    ),
+    Command.Argument(
+        name="private-key-path",
+        description="Path to the file, which stores your private key (in hex representation) for the account. \n"
+        f"Can be used instead of {PRIVATE_KEY_ENV_VAR_NAME} env variable.",
+        type="path",
+    ),
+    Command.Argument(
+        name="signer-class",
+        description="Custom signer class module path.",
+        type="str",
+    ),
+]
+
+
+def create_account_config_from_args(
+    args, chain_id: StarknetChainId
+) -> Optional[AccountConfig]:
+    custom_signer: Optional[BaseSigner] = None
+    if args.account_address is None:
+        return None
+    if args.signer_class:
+        custom_signer = create_custom_signer(args.signer_class)
+        return AccountConfig(account_address=args.account_address, signer=custom_signer)
+    return AccountConfig(
+        account_address=args.account_address,
+        signer=create_stark_curve_signer(
+            account_address=args.account_address,
+            private_key=get_private_key(args.private_key_path),
+            chain_id=chain_id,
+        ),
+    )
+
+
+def create_custom_signer(signer_class_path: str) -> BaseSigner:
+    *module_names, class_name = signer_class_path.split(".")
+    module = ".".join(module_names)
+    signer_module = importlib.import_module(module)
+    # pylint: disable=invalid-name
+    SignerClass = getattr(signer_module, class_name)
+    if not issubclass(SignerClass, BaseSigner):
+        raise InvalidSignerClassException()
+    return SignerClass()
```

Comment:
```suggestion
    signer_factory = getattr(signer_module, class_name)
    
    try:
        signer = signer_factory()
    except TypeError as ex:
        raise InvalidSignerClassException() from ex
        
    if not isinstance(signer, BaseSigner):
        raise InvalidSignerClassException()
        
    return signer
```

1. This way users can pass custom factory functions instead of classes directly.
1. Why `try ... except`?: https://stackoverflow.com/a/5028449/3657395